From 1a7bf6cccebfabee9a38cf53bc53cf5791ae4905 Mon Sep 17 00:00:00 2001
From: Oliver Smith <ollieparanoid@postmarketos.org>
Date: Sun, 8 Jun 2025 13:30:00 +0200
Subject: [PATCH] xiaomi-polaris: remove

The firmware package fails to build, remove the device from stable for
now. We could bring it back later if somebody fixes the package.
---
 device/testing/device-xiaomi-polaris/APKBUILD | 58 -------------------
 .../testing/device-xiaomi-polaris/deviceinfo  | 38 ------------
 .../device-xiaomi-polaris/modules-initfs      |  5 --
 device/testing/device-xiaomi-polaris/phoc.ini |  5 --
 .../30-gpu-firmware.files                     |  3 -
 .../testing/firmware-xiaomi-polaris/APKBUILD  | 42 --------------
 .../firmware-xiaomi-polaris/firmware.files    | 17 ------
 7 files changed, 168 deletions(-)
 delete mode 100644 device/testing/device-xiaomi-polaris/APKBUILD
 delete mode 100644 device/testing/device-xiaomi-polaris/deviceinfo
 delete mode 100644 device/testing/device-xiaomi-polaris/modules-initfs
 delete mode 100644 device/testing/device-xiaomi-polaris/phoc.ini
 delete mode 100644 device/testing/firmware-xiaomi-polaris/30-gpu-firmware.files
 delete mode 100644 device/testing/firmware-xiaomi-polaris/APKBUILD
 delete mode 100644 device/testing/firmware-xiaomi-polaris/firmware.files

diff --git a/device/testing/device-xiaomi-polaris/APKBUILD b/device/testing/device-xiaomi-polaris/APKBUILD
deleted file mode 100644
index af24407846..0000000000
--- a/device/testing/device-xiaomi-polaris/APKBUILD
+++ /dev/null
@@ -1,58 +0,0 @@
-# Maintainer: Molly Sophia <mollysophia379@gmail.com>
-
-# Reference: <https://postmarketos.org/devicepkg>
-pkgname=device-xiaomi-polaris
-pkgdesc="Xiaomi Mi Mix 2S"
-pkgver=3
-pkgrel=1
-url="https://postmarketos.org"
-license="MIT"
-arch="aarch64"
-options="!check !archcheck"
-depends="
-	linux-postmarketos-qcom-sdm845
-	mkbootimg
-	postmarketos-base
-	alsa-ucm-conf-sdm845
-	soc-qcom
-"
-makedepends="devicepkg-dev"
-source="
-	deviceinfo
-	modules-initfs
-	phoc.ini
-"
-subpackages="
-	$pkgname-nonfree-firmware:nonfree_firmware
-	$pkgname-phosh
-"
-
-build() {
-	devicepkg_build $startdir $pkgname
-}
-
-package() {
-	devicepkg_package $startdir $pkgname
-}
-
-nonfree_firmware() {
-	pkgdesc="Modem, WiFi and GPU Firmware, also needed for osk-sdl"
-	depends="
-		firmware-xiaomi-polaris
-		firmware-xiaomi-polaris-initramfs
-		soc-qcom-modem
-	"
-	mkdir "$subpkgdir"
-}
-
-phosh() {
-	install_if="$pkgname=$pkgver-r$pkgrel phosh"
-	install -Dm644 "$srcdir"/phoc.ini \
-		"$subpkgdir"/etc/phosh/phoc.ini
-}
-
-sha512sums="
-de418fb71c9cd1e3e147ff9e4bb4be6c4f507ae6cb6cbf4cc3ec2532c5b7ca293c09f99c46cc09e1fd3233acda7934178b665ebc5052696f1b935ed3fdda5452  deviceinfo
-c1efa00cbfae95c2b0713b4c481f27cba1424b3def1d10ca682424a945adc52435564d917e791f63eda3b6a42acba1453a1c61ab7c15b0dac1678d5e54e1d739  modules-initfs
-e2e417169345c86a5dd34ac0afac4b2b983a755f4c7ba5698c385a01f96b9bd2c3ace1ce5a0569c121675e8b87e88e1cf5c648e995d9c50cba66e52bbc83753c  phoc.ini
-"
diff --git a/device/testing/device-xiaomi-polaris/deviceinfo b/device/testing/device-xiaomi-polaris/deviceinfo
deleted file mode 100644
index a1a7893d01..0000000000
--- a/device/testing/device-xiaomi-polaris/deviceinfo
+++ /dev/null
@@ -1,38 +0,0 @@
-# Reference: <https://postmarketos.org/deviceinfo>
-# Please use double quotes only. You can source this file in shell
-# scripts.
-
-deviceinfo_format_version="0"
-deviceinfo_name="Xiaomi Mi Mix2S"
-deviceinfo_manufacturer="Xiaomi"
-deviceinfo_codename="xiaomi-polaris"
-deviceinfo_year="2018"
-deviceinfo_arch="aarch64"
-deviceinfo_dtb="qcom/sdm845-xiaomi-polaris"
-deviceinfo_append_dtb="true"
-deviceinfo_flash_kernel_on_update="true"
-
-# Device related
-deviceinfo_gpu_accelerated="true"
-deviceinfo_chassis="handset"
-deviceinfo_keyboard="false"
-deviceinfo_external_storage="false"
-deviceinfo_screen_width="1080"
-deviceinfo_screen_height="2160"
-deviceinfo_rootfs_image_sector_size="4096"
-
-# Bootloader related
-deviceinfo_flash_method="fastboot"
-deviceinfo_kernel_cmdline="quiet loglevel=2"
-deviceinfo_generate_bootimg="true"
-deviceinfo_bootimg_qcdt="false"
-deviceinfo_bootimg_dtb_second="false"
-deviceinfo_flash_offset_base="0x00000000"
-deviceinfo_flash_offset_kernel="0x00008000"
-deviceinfo_flash_offset_ramdisk="0x01000000"
-deviceinfo_flash_offset_second="0x00000000"
-deviceinfo_flash_offset_tags="0x00000100"
-deviceinfo_flash_pagesize="4096"
-deviceinfo_flash_sparse="true"
-
-deviceinfo_mesa_driver="msm"
diff --git a/device/testing/device-xiaomi-polaris/modules-initfs b/device/testing/device-xiaomi-polaris/modules-initfs
deleted file mode 100644
index 9578c9b284..0000000000
--- a/device/testing/device-xiaomi-polaris/modules-initfs
+++ /dev/null
@@ -1,5 +0,0 @@
-gpi
-i2c_qcom_geni
-qcom_pmi8998_charger
-qcom_fg
-rmi_i2c
diff --git a/device/testing/device-xiaomi-polaris/phoc.ini b/device/testing/device-xiaomi-polaris/phoc.ini
deleted file mode 100644
index 59213ee6e8..0000000000
--- a/device/testing/device-xiaomi-polaris/phoc.ini
+++ /dev/null
@@ -1,5 +0,0 @@
-# /etc/phosh/phoc.ini
-# Overwrites values set in /usr/share/phosh/phoc.ini
-
-[output:DSI-1]
-scale = 3
diff --git a/device/testing/firmware-xiaomi-polaris/30-gpu-firmware.files b/device/testing/firmware-xiaomi-polaris/30-gpu-firmware.files
deleted file mode 100644
index 010cd6b22c..0000000000
--- a/device/testing/firmware-xiaomi-polaris/30-gpu-firmware.files
+++ /dev/null
@@ -1,3 +0,0 @@
-/lib/firmware/qcom/a630_gmu.bin
-/lib/firmware/qcom/a630_sqe.fw
-/lib/firmware/qcom/sdm845/polaris/a630_zap.mbn
diff --git a/device/testing/firmware-xiaomi-polaris/APKBUILD b/device/testing/firmware-xiaomi-polaris/APKBUILD
deleted file mode 100644
index 22210047b5..0000000000
--- a/device/testing/firmware-xiaomi-polaris/APKBUILD
+++ /dev/null
@@ -1,42 +0,0 @@
-# Maintainer: Molly Sophia <mollysophia379@gmail.com>
-
-pkgname=firmware-xiaomi-polaris
-pkgver=0.1
-pkgrel=1
-pkgdesc="Firmware for Xiaomi Mi Mix2s"
-url="https://github.com/MollySophia/firmware-xiaomi-polaris"
-arch="aarch64"
-depends="linux-firmware-qcom linux-firmware-qca linux-firmware-ath10k"
-license="proprietary"
-options="!check !strip !archcheck !tracedeps pmb:cross-native"
-
-_version="ec608e1a8bf2d4aded30abed8476f805fd6f6e3c"
-
-source="$pkgname.tar.gz::$url/archive/$_version.tar.gz
-	firmware.files
-	30-gpu-firmware.files"
-subpackages="
-	$pkgname-initramfs:firmware_initramfs"
-
-package() {
-	cd "$srcdir/$pkgname-$_version/"
-	for i in $(cat "$srcdir"/firmware.files); do
-		install -Dm644 "$i" "$pkgdir/$i"
-	done
-}
-
-firmware_initramfs() {
-	pkgdesc="Files to be included in the initramfs to support osk-sdl"
-	depends="$pkgname"
-	install_if="$pkgname-nonfree-firmware"
-	mkdir "$subpkgdir"
-
-	install -Dm644 "$srcdir/30-gpu-firmware.files" \
-		"$subpkgdir/usr/share/mkinitfs/files/30-gpu-firmware.files"
-}
-
-sha512sums="
-9e9d095d4c2469acd22325c1546b53437557362cd90c407b9f212f9ac168ef577d5fb9b534b0b4eaa7f3da221d13ca4f488aa9364a34d6c2490aeac5e79145ee  firmware-xiaomi-polaris.tar.gz
-2a172fff3844c392c3f03c6d7f1508e03ed18fe916261cedebde79bb38fc21aaf6ed02e5cb88aa3546ea7b978717dc15d0bc71f64703d848cb1d974756dd3670  firmware.files
-7d430240abae210dd28db688258b45d025915db6fe60345346a1d2d12d7f5bddb0755b3bed8689f54081b7a5188d04906b95c1ea6bbf599019e54544bde1210b  30-gpu-firmware.files
-"
diff --git a/device/testing/firmware-xiaomi-polaris/firmware.files b/device/testing/firmware-xiaomi-polaris/firmware.files
deleted file mode 100644
index 912d7c2823..0000000000
--- a/device/testing/firmware-xiaomi-polaris/firmware.files
+++ /dev/null
@@ -1,17 +0,0 @@
-./lib/firmware/tas2559_uCDSP.bin
-./lib/firmware/postmarketos/ath10k/WCN3990/hw1.0/board-2.bin
-./lib/firmware/postmarketos/qca/crbtfw21.tlv
-./lib/firmware/qcom/sdm845/polaris/adspr.jsn
-./lib/firmware/qcom/sdm845/polaris/adspua.jsn
-./lib/firmware/qcom/sdm845/polaris/cdspr.jsn
-./lib/firmware/qcom/sdm845/polaris/modemr.jsn
-./lib/firmware/qcom/sdm845/polaris/modemuw.jsn
-./lib/firmware/qcom/sdm845/polaris/venus.mbn
-./lib/firmware/qcom/sdm845/polaris/wlanmdsp.mbn
-./lib/firmware/qcom/sdm845/polaris/a630_zap.mbn
-./lib/firmware/qcom/sdm845/polaris/adsp.mbn
-./lib/firmware/qcom/sdm845/polaris/cdsp.mbn
-./lib/firmware/qcom/sdm845/polaris/ipa_fws.mbn
-./lib/firmware/qcom/sdm845/polaris/mba.mbn
-./lib/firmware/qcom/sdm845/polaris/modem.mbn
-./lib/firmware/qca/polaris/crnv21.bin
\ No newline at end of file
-- 
2.49.0

